extensions/eclipse-che-theia-plugin-ext/src/browser/che-product-main.ts

```diff
@@ -0,0 +1,42 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { interfaces } from 'inversify';
+import { CheProduct, CheProductMain, PLUGIN_RPC_CONTEXT, CheProductService, ProductInfo } from '../common/che-protocol';
+import { RPCProtocol } from '@theia/plugin-ext/lib/common/rpc-protocol';
+
+export class CheProductMainImpl implements CheProductMain {
+
+    private proxy: CheProduct;
+
+    private readonly cheProductService: CheProductService;
+
+    private productInfo: ProductInfo;
+
+    constructor(container: interfaces.Container, rpc: RPCProtocol) {
+        this.proxy = rpc.getProxy(PLUGIN_RPC_CONTEXT.CHE_PRODUCT);
+        this.cheProductService = container.get(CheProductService);
+        this.initialize();
+    }
+
+    async initialize(): Promise<void> {
+        this.productInfo = await this.cheProductService.getProductInfo();
+
+        // Temporary solution.
+        // We should wait for some time until proxy is being fully initialized.
+        setTimeout(() => {
+            this.proxy.$setName(this.productInfo.name);
+            this.proxy.$setLogo(this.productInfo.logo);
+            this.proxy.$setDescription(this.productInfo.description);
+            this.proxy.$setLinks(this.productInfo.links);
+        }, 500);
```

Comment:
what is the expected proper solution ? it looks something that is not working when we're using timeout

Reply:
I still don't get it there and why we have setTimeout as it'll mostly fail.

we have async everywhere so it should either be async or deleted/reworked

Reply:
Main part should send values to the plugin. In our plugin system main part is initialized earlier than plugin part. So, if we send the values immediately, nobody receive it.

Reply:
`che.product` plugin API has only variables. They have to be initialized somehow before using.
We cannot use asynchronous method as you requested to use synchronous variables.